compiler/rustc_errors/src/diagnostic.rs

```diff
@@ -939,6 +939,23 @@ impl Diagnostic {
         self
     }
 
+    /// Add a subdiagnostic from a type that implements `Subdiagnostic` (see
+    /// [rustc_macros::Subdiagnostic]). Performs eager translation of any translatable messages
+    /// used in the subdiagnostic, so suitable for use with repeated messages (i.e. re-use of
+    /// interpolated variables).
+    pub fn eager_subdiagnostic(
```

Comment:
Can I expect a use case of this function?

Reply:
See 113e94369cb72a98648c12c263475821bbff7d9c